Apps/WebClient/src/ClientApp/app/components/timeline/commentSection.vue

```diff
@@ -0,0 +1,174 @@
+          <b-form @submit.prevent="addComment">
+            <b-form-input
+              type="text"
+              autofocus
+              class="newComment"
+              v-model="newComment"
+              placeholder="Enter a comment"
+              maxlength="1000"
+            ></b-form-input>
+          </b-form>
+        </b-collapse>
+      </b-col>
+    </b-row>
+    <b-row>
+      <b-col>
+        <b-collapse :visible="showComments">
+          <div v-if="!this.isLoadingComments">
+            <div v-for="comment in this.comments" :key="comment.id">
+              <Comment :comment="comment"></Comment>
+            </div>
+          </div>
+        </b-collapse>
+      </b-col>
+    </b-row>
+  </div>
+</template>
+<script lang="ts">
+import Vue from "vue";
+import UserComment from "@/models/userComment";
+import CommentComponent from "@/components/timeline/comment.vue";
+import MedicationTimelineEntry from "@/models/medicationTimelineEntry";
+import User from "@/models/user";
+import { Prop, Component } from "vue-property-decorator";
+import { IUserCommentService } from "@/services/interfaces";
+import { SERVICE_IDENTIFIER } from "@/plugins/inversify";
+import container from "@/plugins/inversify.config";
+import {
+  faCommentAlt,
+  IconDefinition,
+} from "@fortawesome/free-solid-svg-icons";
+import { Getter } from "vuex-class";
+
+@Component({
+  components: {
+    Comment: CommentComponent,
+  },
+})
+export default class CommentSectionComponent extends Vue {
+  @Getter("user", { namespace: "user" }) user!: User;
+  @Prop() parentEntry!: MedicationTimelineEntry;
+  private commentService!: IUserCommentService;
+  private showComments: boolean = false;
+  private showInput: boolean = false;
+  private isCommentSaving: boolean = false;
+  private isLoadingComments: boolean = false;
+  private newComment: string = "";
+  private comments: UserComment[] = [];
+  private numComments = 0;
+  private hasErrors: boolean = false;
+
+  private mounted() {
+    this.commentService = container.get<IUserCommentService>(
+      SERVICE_IDENTIFIER.UserCommentService
+    );
+    this.getComments();
+  }
+
+  private get hasComments(): boolean {
+    return this.comments.length > 0;
+  }
+
+  private get commentIcon(): IconDefinition {
+    return faCommentAlt;
+  }
+
+  private sortComments() {
+    this.comments.sort((a, b) => {
+      if (a.createdDateTime > b.createdDateTime) {
+        return -1;
+      } else if (a.createdDateTime < b.createdDateTime) {
+        return 1;
+      } else {
+        return 0;
+      }
+    });
+  }
+
+  private toggleComments(): void {
+    this.showComments = !this.showComments;
+  }
+
+  private toggleCommentInput(): void {
+    this.showInput = !this.showInput;
+    this.newComment = "";
+  }
+
+  private addComment(): void {
+    this.isCommentSaving = true;
+    let commentPromise = this.commentService
+      .createComment({
+        text: this.newComment,
+        parentEntryId: this.parentEntry.id,
```

Comment:
Type error: parentEntryId is a number while this.parentEntry.id is a string. 
parentEntryId should be a string.